feat(curriculum): add English curriculum block 3.2 hobbies and interests

miyaliu666 opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->


<!-- Feel free to add any additional description of changes below this line -->

miyaliu666 committed:
feat: created dialogue 1 tasks

miyaliu666 committed:
fix: remove option numbers and add backticks for dialogue 1

miyaliu666 committed:
feat: created dialogue 2 tasks

miyaliu666 committed:
fix: remove backticks in feedbacks or add backticks in answers for dialogue 2

miyaliu666 committed:
feat: created dialogue3 tasks

miyaliu666 committed:
change the type of task 54 to 22

miyaliu666 committed:
created tasks for dialogue 4

miyaliu666 committed:
feat: created tasks for dialogue 5

miyaliu666 committed:
Merge branch 'freeCodeCamp:main' into main

scissorsneedfoodtoo committed:
Merge branch 'main' into feat-3.2

scissorsneedfoodtoo committed:
chore: rename files with rename challenge files tool

scissorsneedfoodtoo committed:
fix: add blank lines between code blocks

scissorsneedfoodtoo committed:
fix: manually rename files and set ids for tasks 54 and 68

scissorsneedfoodtoo committed:
fix: meta title for dialogue 3

scissorsneedfoodtoo commented:
Hey @miyaliu666, hope you don't mind, but I went in and used Naomi's script to rename the files for this block, and fixed some other errors. That should make it easier to review your PR in the future.

miyaliu666 commented:
> Hey @miyaliu666, hope you don't mind, but I went in and used Naomi's script to rename the files for this block, and fixed some other errors. That should make it easier to review your PR in the future.

@scissorsneedfoodtoo thanks for helping rename them and fix errors:)

miyaliu666 committed:
fix: add a line in tasks 48, 58 and 60

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/65800cf36faba0fbfa1027b6.md:
This is a good opportunity to teach negative questions. I suggest editing the description to focus on teaching this.

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/65800df7fc5d49fcd7209248.md:
```diff
@@ -0,0 +1,39 @@
+audioPath: curriculum/js-music-player/We-Are-Going-to-Make-it.mp3
+---
+
+---
+<!--
+AUDIO REFERENCE:
+Sophie: Well, why don't you try it? I'm planning to go rafting this weekend.
+-->
```
```suggestion
<!--
AUDIO REFERENCE:
Sophie: Well, why don't you try it? I'm planning to go rafting this weekend.
-->
```

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/65800e60dad5fefd4ed91589.md:
```diff
@@ -0,0 +1,50 @@
+
+# --description--
+
+Following the structure `I'm planning to + base form + to + -ing`, the sentence indicates a future intention to engage in an activity.
```
```suggestion
Following the structure `I'm planning to` + base form + `to` + `-ing`, the sentence indicates a future intention to engage in an activity.
```

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/65800e60dad5fefd4ed91589.md:
I suggest a review to decide whether this is going to be type 19 or 22.

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/65801182280f63ff10ca4d4f.md:
```diff
@@ -0,0 +1,46 @@
+
+### --feedback--
+
+This word is the third person singular present tense of 'sound', used to indicate how something seems from the statements or actions made.
```
```suggestion
This word is the third person singular present tense of `sound`, used to indicate how something seems from the statements or actions made.
```

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/658011ef9ec114ff80ce5e42.md:
I'd add the audio and change the description to "listen to the audio and answer the question"

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/65802ee9706eb103aea442f8.md:
```diff
@@ -0,0 +1,50 @@
+
+## --text--
+
+Choose the correct expanded form of the contraction `I'd` in the given context.
+
+`Really? That sounds great! _.`
```
```suggestion
`Really? That sounds great! ...`
```
I'm still not 100% sure this is the best approach but this could be a way to fix this hybrid task.

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/65802ee9706eb103aea442f8.md:
```diff
@@ -0,0 +1,50 @@
+
+# --description--
+
+The phrase `I'd love to join` is used with the contraction `I'd`, which is short for `I would`, expressing willingness and enthusiasm.
```
```suggestion
Choose the correct expanded form of the contraction `I'd` that could conclude the following sentence:
```

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/658030876ac4f605145aeae1.md:
```diff
@@ -0,0 +1,15 @@
+
+Watch the video above to understand the context of the upcoming lessons.
+
+## --assignment--
```
```suggestion
# --assignment--
```

nieldakarla reviewed (changes requested):
Hey Miya, thank you so much for your hard work with this block. It is huge! 🔥  

I've added suggestions to most challenges. Some changes I've added myself others I created comments with instructions. If there are any that you disagree or don't know how to implement, please feel free to reach out. 

One of the reasons why I've added so many comments is that I believe this is a good opportunity for us to fine-tune what we want from the English Curriculum challenges and how they should look. If there's any suggestion you feel you can't do let me know.

Again, congratulations for the great work you've done! 😃

moT01 commented:
> One of the reasons why I've added so many comments is that I believe this is a good opportunity for us to fine-tune what we want from the English Curriculum challenges and how they should look.

This is the time to do that @nieldakarla. We likely won't make many more changes to these before they go live.

miyaliu666 committed:
Merge branch 'feat-3.2' of github.com:miyaliu666/freeCodeCamp into feat-3.2

miyaliu666 commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657d6008a37af78962d30873.md:
```diff
@@ -0,0 +1,38 @@
+
+### --feedback--
+
+This blank is for the past participle form of the main verb in the present perfect tense.
```
Do we want to show the answer in feedback?

miyaliu666 commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657d9af10739f78d5401ad1d.md:
```diff
@@ -0,0 +1,48 @@
+
+The phrase `As a hobby` is used with the preposition `as`. Choose the correct preposition to complete the sentence.
+
+# --question--
+
+## --text--
+
+`Bob enjoys sketching and painting _ his hobby.`
+
+## --answers--
+
+`on`
+
+### --feedback--
+
+`On` is not the correct preposition to convey engagement in an activity.
+
+---
+
+`for`
+
+### --feedback--
+
+`For` implies doing something for a purpose, which is not the focus in this context.
+
+---
+
+`as`
+
+---
+
+`with`
+
+### --feedback--
+
+`With` does not convey the idea of pursuing an activity for personal enjoyment.
+
+## --video-solution--
+
+3
```
The forth option is using "as" appropriately. I suggest:

`I use my phone as to make calls.`

### --feedback---

This sentence is incorrect because `as` is not needed with `to` in this context. The correct sentence should be: `I use my phone to make calls.`

miyaliu666 commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657d9dbe3f6a1b90f79d81f2.md:
Task 16 is filling blanks with this phrase, so I make this one a comprehension practice with audio from the dialogue.

miyaliu666 commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657da2e3cf29239535791805.md:
```diff
@@ -0,0 +1,15 @@
+
+Watch the video above to understand the context of the upcoming lessons.
+
+## --assignment--
```
I test creating a dialogue type challenge again - there is `##` before "assignment".

nieldakarla replied:
When we create the assignment with two ## they look like this: 
![image](https://github.com/freeCodeCamp/freeCodeCamp/assets/90112119/65cc1e6f-81c9-42fa-b261-5a763baf1e17)
I don't know why it is creating the challenges with two ## but as far as I know there should be just one.

miyaliu666 commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657e7a17f47a8fc2d037f48e.md:
```diff
@@ -0,0 +1,54 @@
+
+# --description--
+
+The expression `one of these days` implies a willingness to engage in an activity without specifying a definite time. It often suggests spontaneity.
```
You didn't give a simpler explanation here.

miyaliu666 committed:
Apply suggestions for tasks 1-36 from code review

Co-authored-by: Nielda Karla <90112119+nieldakarla@users.noreply.github.com>

miyaliu666 committed:
update tasks based on review suggestions

miyaliu666 committed:
Merge branch 'feat-3.2' of github.com:miyaliu666/freeCodeCamp into feat-3.2

miyaliu666 commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657fbde9a43e35ec1ebafe56.md:
We already taught `Would you like to come`?

miyaliu666 committed:
fix: update tasks based on review suggestions

miyaliu666 committed:
Apply suggestions from code review

Co-authored-by: Nielda Karla <90112119+nieldakarla@users.noreply.github.com>

miyaliu666 commented:
Hi @nieldakarla  I've updated tasks based on your review suggestions. Could you re-review them? Thanks.

nieldakarla committed:
Apply suggestions from code review

After discussing these changes with Miya, we decided to add these changes.

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657d9eed9f6a5a91c7c5f210.md:
```diff
@@ -0,0 +1,46 @@
+AUDIO REFERENCE: 
+Bob: I like landscapes and nature. It's incredible to capture the beauty of our world in pictures.
+-->
+
+# --description--
+
+---
+# --description--
+
```
```suggestion

# --description--

```
Description was repeated. Sorry for that.

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657da2e3cf29239535791805.md:
```diff
@@ -0,0 +1,15 @@
+
+Watch the video above to understand the context of the upcoming lessons.
+
+## --assignment--
```
```suggestion
# --assignment--
```

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657e758778ff8dc01567c9e7.md:
````diff
@@ -13,13 +13,13 @@ Jake: That's great! How long have you been into cycling?
 # --description--
 
-To form questions with the Present Perfect Continuous Tense, you need to rearrange the sentence structure. This involves changing the position of the subject, the auxiliary verb `have been`, and the present participle of the main verb. In this dialogue, Jake inquires about the duration of someone's enthusiasm for cycling.
+Remember, to form questions with the Present Perfect Continuous Tense, you need to rearrange the sentence structure. This involves changing the position of the subject, the auxiliary verb `have been`, and the present participle of the main verb. ```
````
```suggestion
Remember, to form questions with the Present Perfect Continuous Tense, you need to rearrange the sentence structure. This involves changing the position of the subject, the auxiliary verb `have been`, and the present participle of the main verb.
```

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657e76356b6af6c07fe338c1.md:
```diff
@@ -0,0 +1,34 @@
+AUDIO REFERENCE: 
+Linda: I've been a cyclist for more than 15 years. It's a love that hasn’t disappeared with time.
+-->
+
+# --description--
+
+---
+# --description--
+
```
```suggestion

# --description--

```

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657e77615bb8dec146e91f96.md:
```diff
@@ -0,0 +1,54 @@
+id: 657e77615bb8dec146e91f96
+title: Task 30
+challengeType: 22
+dashedName: task-30
+---
```
```suggestion
dashedName: task-30
audioPath: curriculum/js-music-player/We-Are-Going-to-Make-it.mp3
---
```

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657e77615bb8dec146e91f96.md:
```diff
@@ -0,0 +1,54 @@
+
+## --sentence--
+
+`People say we never forget. Say, do you still have a bike? _ _ _ _ go cycling with me one of these days?`
```
```suggestion
` _ _ _ _ go cycling with me one of these days?`
```

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657ec6ac4de4eac8bab2f2a7.md:
```diff
@@ -0,0 +1,38 @@
+---
+id: 657ec6ac4de4eac8bab2f2a7
+title: Task 39
+challengeType: 19
```
```suggestion
challengeType: 22
```
The challenge type was wrong and returning an error msg.

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657ee66dd1216ad6c61f1168.md:
```diff
@@ -0,0 +1,34 @@
+---
+id: 657ee66dd1216ad6c61f1168
+title: Task 52
+challengeType: 19
```
```suggestion
challengeType: 22
```

nieldakarla commented on curriculum/challenges/english/21-a2-english-for-developers/learn-how-to-talk-about-hobbies-and-interests/657ef4e5a4a1e1e126eba6dd.md:
Challenges with the entire Dialogues 3 and 5 are out of order. I'll attempt to fix that directly on the meta.json file:
![image](https://github.com/freeCodeCamp/freeCodeCamp/assets/90112119/7db8797d-bc11-4f46-9ec4-0cea541de880)

nieldakarla commented on curriculum/challenges/_meta/learn-how-to-talk-about-hobbies-and-interests/meta.json:
```diff
@@ -10,8 +10,444 @@
+    },
+    {
+      "id": "65800e60dad5fefd4ed91589",
+      "title": "Task 97"
+    },
+    {
+      "id": "65800f4d194382fdebb81e1f",
+      "title": "Task 98"
+    },
+    {
+      "id": "658010478daa16fe79d8113a",
+      "title": "Task 99"
+    },
+    {
+      "id": "65801182280f63ff10ca4d4f",
+      "title": "Task 100"
+    },
+    {
+      "id": "658011ef9ec114ff80ce5e42",
+      "title": "Task 101"
+    },
+    {
+      "id": "658013bd3b1a06001a59e006",
+      "title": "Task 102"
+    },
+    {
+      "id": "65802ee9706eb103aea442f8",
+      "title": "Task 103"
+    },
+    {
+      "id": "65802f717cef8c042af950b8",
+      "title": "Task 104"
+    },
+    {
+      "id": "65802fe92ef0f404ba0437f7",
+      "title": "Task 105"
+    },
+    {
+      "id": "658030876ac4f605145aeae1",
+      "title": "Dialogue 5: Getting into Extreme Sports"
```
```suggestion
      "id": "657ec6ac4de4eac8bab2f2a7",
      "title": "Task 39"
    },
    {
      "id": "657ef0061e99a3ddd0e1245f",
      "title": "Dialogue 3: The Collector"
    },
    {
      "id": "657eda6e48e0d7c92f0af163",
      "title": "Task 40"
    },
    {
      "id": "657edbc3e12e35cfc1d80358",
      "title": "Task 41"
    },
    {
      "id": "657edc5136791ed04ffab234",
      "title": "Task 42"
    },
    {
      "id": "657ee0a0c52d0ed1164a85d8",
      "title": "Task 43"
    },
    {
      "id": "657ee214b9ad1ad2b6f8325c",
      "title": "Task 44"
    },
    {
      "id": "657ee25acee11cd3122e0876",
      "title": "Task 45"
    },
    {
      "id": "657ee2f44346f3d4017cc990",
      "title": "Task 46"
    },
    {
      "id": "657ee3954d64e5d465500620",
      "title": "Task 47"
    },
    {
      "id": "657ee4171371e9d4d1402e91",
      "title": "Task 48"
    },
    {
      "id": "657ee47c1af836d5289254d9",
      "title": "Task 49"
    },
    {
      "id": "657ee51ce8dac8d5a227f758",
      "title": "Task 50"
    },
    {
      "id": "657ee5ddb47b1dd62f87d784",
      "title": "Task 51"
    },
    {
      "id": "657ee66dd1216ad6c61f1168",
      "title": "Task 52"
    },
    {
      "id": "657ee781258676d764a5282f",
      "title": "Task 53"
    },
    {
      "id": "657ee8a9a195b8d8756ca5e1",
      "title": "Task 54"
    },
    {
      "id": "657ee957cb4719d9031a0be4",
      "title": "Task 55"
    },
    {
      "id": "657ee9b69c2e26d964f67ee4",
      "title": "Task 56"
    },
    {
      "id": "657eeb163e1993d9e342d661",
      "title": "Task 57"
    },
    {
      "id": "657eebaa042b5cda6ec2dac9",
      "title": "Task 58"
    },
    {
      "id": "657eec0a21bc7adada04453e",
      "title": "Task 59"
    },
    {
      "id": "657eecb3b368badb3cc7fe4c",
      "title": "Task 60"
    },
    {
      "id": "657eed5ed318e4dbbce6903a",
      "title": "Task 61"
    },
    {
      "id": "657ef4e5a4a1e1e126eba6dd",
      "title": "Dialogue 4: Inviting to A Convention"
    },
    {
      "id": "657ef58855ff45e1e4ca70ee",
      "title": "Task 62"
    },
    {
      "id": "657ef61be0a682e2591776f6",
      "title": "Task 63"
    },
    {
      "id": "657fb48f31654ae4cb52d395",
      "title": "Task 64"
    },
    {
      "id": "657fb5034da144e55cbee9d7",
      "title": "Task 65"
    },
    {
      "id": "657fb5afeeba2de5d01dda0e",
      "title": "Task 66"
    },
    {
      "id": "657fb665a39478e642f5a139",
      "title": "Task 67"
    },
    {
      "id": "657fb8cfbd0e4ae797fc6077",
      "title": "Task 68"
    },
    {
      "id": "657fb92c6f888fe8013f1a28",
      "title": "Task 69"
    },
    {
      "id": "657fb980a9b567e860b77f2c",
      "title": "Task 70"
    },
    {
      "id": "657fba163fec41e8e00c5817",
      "title": "Task 71"
    },
    {
      "id": "657fbad88163d8e96189f823",
      "title": "Task 72"
    },
    {
      "id": "657fbb3a1a63d0e9c307fd83",
      "title": "Task 73"
    },
    {
      "id": "657fbc14d41a20ea5f286378",
      "title": "Task 74"
    },
    {
      "id": "657fbc85c25a16eac8356182",
      "title": "Task 75"
    },
    {
      "id": "657fbcecfc4d42eb28700349",
      "title": "Task 76"
    },
    {
      "id": "657fbd591461eaeba07ffff1",
      "title": "Task 77"
    },
    {
      "id": "657fbde9a43e35ec1ebafe56",
      "title": "Task 78"
    },
    {
      "id": "657fff0bfb6a28f1d70fa9ef",
      "title": "Task 79"
    },
    {
      "id": "657fff7dabba2ff23993b08c",
      "title": "Task 80"
    },
    {
      "id": "6580001adc7fd4f2b244f3a5",
      "title": "Task 81"
    },
    {
      "id": "65800082405352f30c6dbc7a",
      "title": "Task 82"
    },
    {
      "id": "65800148406738f397561d77",
      "title": "Task 83"
    },
    {
      "id": "658001a018bbcbf3fd84f832",
      "title": "Task 84"
    },
    {
      "id": "6580026241ae0ef46b181e49",
      "title": "Task 85"
    },
    {
      "id": "658002d23e245ff4ca8542d5",
      "title": "Task 86"
    },
    {
      "id": "658003870281a9f5541085af",
      "title": "Task 87"
    },
    {
      "id": "658003d8ff9da6f5c08971a1",
      "title": "Task 88"
    },
     {
      "id": "658030876ac4f605145aeae1",
      "title": "Dialogue 5: Getting into Extreme Sports"
      },
    {
      "id": "658009d86dc9caf988e2ea64",
      "title": "Task 89"
    },
    {
      "id": "65800a95389cc0fa4c197587",
      "title": "Task 90"
    },
    {
      "id": "65800b1c13005dfa9df75d73",
      "title": "Task 91"
    },
    {
      "id": "65800b96989013fb24aa1b70",
      "title": "Task 92"
    },
    {
      "id": "65800c12978ba7fb82007446",
      "title": "Task 93"
    },
    {
      "id": "65800cf36faba0fbfa1027b6",
      "title": "Task 94"
    },
    {
      "id": "65800d61890343fc5cce0ec8",
      "title": "Task 95"
    },
    {
      "id": "65800df7fc5d49fcd7209248",
      "title": "Task 96"
    },
    {
      "id": "65800e60dad5fefd4ed91589",
      "title": "Task 97"
    },
    {
      "id": "65800f4d194382fdebb81e1f",
      "title": "Task 98"
    },
    {
      "id": "658010478daa16fe79d8113a",
      "title": "Task 99"
    },
    {
      "id": "65801182280f63ff10ca4d4f",
      "title": "Task 100"
    },
    {
      "id": "658011ef9ec114ff80ce5e42",
      "title": "Task 101"
    },
    {
      "id": "658013bd3b1a06001a59e006",
      "title": "Task 102"
    },
    {
      "id": "65802ee9706eb103aea442f8",
      "title": "Task 103"
    },
    {
      "id": "65802f717cef8c042af950b8",
      "title": "Task 104"
    },
    {
      "id": "65802fe92ef0f404ba0437f7",
      "title": "Task 105"
```

Tasks for dialogue 3 and 5 were out of place. I've made this change so they appear in the correct order.

nieldakarla reviewed (changes requested):
Great work @miyaliu666 ! Thank you for thoughtfully considering all my suggestions. I've found that the challenges for Dialogue 3 and 5 were out of place and added a fix to that. Besides this, only a few minor fixes.

nieldakarla committed:
Fix: Reordered type 21 challenges dialogues 3 and 5

Challenges type 21 for Dialogues 3 and 5 were out of place, I've re-ordered them on meta.json. I took the opportunity and also fixed some smaller details. I've left individual comments for those.

nieldakarla reviewed (approved):
Now, I believe we are ready to go with this one! 🎉🎉🎉

DanielRosa74 reviewed (approved):
LGTM

ojeytonwilliams reviewed (approved):
This all looks good from a platform perspective.
